### ISSUE TITLE
import/no-extraneous-dependencies: Complains about node standard libraries

### ISSUE THREAD

**Issue by burimshala** [benmosher/eslint-plugin-import#1396]
It throws an error which says `'path' should be listed in the project's dependencies. Run 'npm i -S path' to add it`.

The eslintrc contains environment configuration:

```
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "jest": true
  },
```

It does not throw this error with eslint 5.x.x only with 6.x.x

**Comment by ljharb** [comment]
I wonder if perhaps eslint 6 handles path differently with the node env?

Either way we should use the resolve module’s isCore method for this.

**Comment by dkrnl** [comment]
i catch similar bug

**Comment by AlekseyLeshko** [comment]
Try set **packageDir** for **import/no-extraneous-dependencies** rule.
[More info in the article](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md)
It’s work for me.

**Comment by taion** [comment]
There does appear to have been a change to resolution. With ESLint v6, e.g. `path` now seems to resolve to `path-browserify`, rather than the Node built-in.

**Comment by ljharb** [comment]
If so that’s a major bug; can you provide a minimal repro?

**Comment by ljharb** [comment]
Also can you confirm if eslint is depending on `resolve`, and if so, can you try downgrading it to v1.11.0?

**Comment by taion** [comment]
I don't think it's `resolve` or `require.resolve`. I think it's something specific to the webpack resolver now.

If I log the inputs to `isBuiltIn` in: https://github.com/benmosher/eslint-plugin-import/blob/65121104ea4659f81e8cb07184b7c1ee346a4a6f/src/core/importType.js#L25

I see e.g.

```
path {
  'import/resolver': { webpack: { config: [Object] }, node: { extensions: [Array] } },
  react: { pragma: 'React', version: '16.0' },
  propWrapperFunctions: [ 'forbidExtraProps', 'exact', 'Object.freeze' ],
  'import/extensions': [ '.js', '.mjs', '.jsx' ],
  'import/core-modules': [],
  'import/ignore': [ 'node_modules', '\\.(coffee|scss|css|less|hbs|svg|json)$' ]
} /Users/jjia/vendor/react-bootstrap/node_modules/path-browserify/index.js
```

I believe this issue comes from `eslint-module-utils/resolve`.

**Comment by ljharb** [comment]
I mean, it makes perfect sense to methat the webpack resolver maps core modules to their respective browser shims - but i don’t know why eslint 6 would change that.

**Comment by dylang** [comment]
FWIW, I'm seeing this error in a project using eslint 5.

It appears to be using the `import/resolver`: `eslint-import-resolver-webpack` instead of `node`, but I haven't figured out why.

**Comment by kelchm** [comment]
I'm seeing the same issue after upgrading to eslint v6.

I can't explain _why_ it works, but try adding this to your eslint config:
```
{
...
  settings: {
    'import/resolver': {
      node: {},
    },
  }
}
```

**Comment by MaximDevoir** [comment]
Adding onto [@kelchm 's solution](https://github.com/benmosher/eslint-plugin-import/issues/1396#issuecomment-509384041), `node` had to be the first resolver for my eslint to work.

```js
...
  settings: {
    'import/resolver': {
      node: {}, // placed above other resolver configs
      webpack: { ... }
    }
  }
}
```

**Comment by taion** [comment]
The above worked for me as well. We had a repo where most files ran through webpack, but a few were Node configs. Overriding those to use the `node` resolver fixed the issues.

As such, I don't think there was a bug here.

**Comment by ryansully** [comment]
The `eslint-import-resolver-node` package would need to be installed to devDependencies in your project's package.json in order to set the `node` property, right?

**Comment by taion** [comment]
Correct.

**Comment by Xotic750** [comment]
If I revert back from eslint@6.1.0 to @5.16.0, while leaving eslint-plugin-import@2.18.2 then the problem goes away.

**Comment by brendonco** [comment]
> eslint-import-resolver-node

doesnt seem to work

**Comment by ljharb** [comment]
https://github.com/benmosher/eslint-plugin-import/issues/1396#issuecomment-511007063 is the proper solution.

Please file a new issue if you're still having trouble with the latest release.

**Comment by AakashAlagarsamy** [comment]
I'm using eslint 7.29.0
Same error is showing

'webpack-merge' should be listed in the project's dependencies, not devDependencies.eslintimport/no-extraneous-dependencies

My file is below

**development.config.js**
```
const { merge } = require('webpack-merge');
const webpack = require('webpack');
```

**Comment by ljharb** [comment]
@AakashAlagarsamy and is your eslint configured to allow dev deps in that file?
